docs: Resolve `Inline interpreted text [...] without end-string` warnings

paulinebanye opened the pull request:
### **Description**
This PR addresses the `Inline interpreted text or phrase reference start-string without end-string` warnings encountered during the Sphinx `make html` build process (Issue #1671).
These warnings occurred because of a reference that starts with a backtick but doesn't have a matching closing backtick in the docstring.

```python   
# report.py
~/_comparison/report.py:docstring of skore.sklearn._comparison.report.ComparisonReport:4: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
~/_comparison/report.py:docstring of skore.sklearn._comparison.report.ComparisonReport:4: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]

# metrics_accessor.py
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.accuracy:23: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]       
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.brier_score:23: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]    
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.custom_metric:47: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]  
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.log_loss:23: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]       
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.precision:50: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]      
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.r2:32: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.recall:51: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.report_metrics:50: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils] 
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.rmse:32: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.roc_auc:56: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]        
~/_comparison/metrics_accessor.py:docstring of skore.sklearn._comparison.metrics_accessor._MetricsAccessor.timings:11: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
```
---

### **Environment**

- Model: Dell Latitude E7440
- RAM: 12GB 
- OS: Windows 10 pro
- Python Version: Python 3.12
- Terminal: Ubuntu 22.04 LTS

---

### **Steps to reproduce:**
1. Clone the skore repository
2. Create and activate a virtual environment using the Ubuntu terminal
3. Run the _make install-skore_ command.
4. Change directory into the sphinx directory
5. Run `make html`.
---

### **Solution:**
- Changed **:class:~skore.EstimatorReport`s** where the **s** was outside the backticks in the metrics_accessor.py file, to **:class:~skore.EstimatorReport(s)**

  *_before_*
  ```
  # metrics_accessor.py
         **kwargs : dict
              Any additional keyword arguments to be passed to the metric function.

          aggregate : {"mean", "std"}, list of such str or None, default=("mean", "std")
              Function to aggregate the scores across the cross-validation splits.
              None will return the scores for each split.
              Ignored when comparison is between :class:`~skore.EstimatorReport`s.
  ```

  *_after_*
  ```
  # metrics_accessor.py
         **kwargs : dict
              Any additional keyword arguments to be passed to the metric function.
  
          aggregate : {"mean", "std"}, list of such str or None, default=("mean", "std")
              Function to aggregate the scores across the cross-validation splits.
              None will return the scores for each split.
              Ignored when comparison is between :class:`~skore.EstimatorReport(s)`.
  ```

- Changed **:class:~skore.EstimatorReport`s** where the **s** was outside the backticks in the report.py file, to **:class:skore.EstimatorReport reports**.
 
  *_before_*
  ```
  # report,py
  class ComparisonReport(_BaseReport, DirNamesMixin):
      """Report for comparing reports.

      This object can be used to compare several :class:`skore.EstimatorReport`s,
      or several :class:`~skore.CrossValidationReport`s.
  ```

  *_after_*
  ```
  # report,py
  class ComparisonReport(_BaseReport, DirNamesMixin):
      """Report for comparing reports.

      This object can be used to compare several :class:`skore.EstimatorReport` reports,
      or several :class:`~skore.CrossValidationReport` reports.
  ```

_Note:_
I was unsure which of the modifications would be preferred - **class:\`skore.EstimatorReport\` reports** or **class:\`~skore.EstimatorReport(s)\`**, so I left both. 


cc @glemaitre @thomass-dev @auguste-probabl

thomass-dev commented on skore/src/skore/sklearn/_comparison/metrics_accessor.py:
```diff
@@ -112,7 +112,7 @@ def report_metrics(
             Function to aggregate the scores across the cross-validation splits.
             None will return the scores for each split.
-            Ignored when comparison is between :class:`~skore.EstimatorReport`s.
+            Ignored when comparison is between :class:`~skore.EstimatorReport(s)`.
```
IMO, you should change for:

```suggestion
            Ignored when comparison is between :class:`~skore.EstimatorReport` instances.
```

This is better than `~skore.EstimatorReport reports` which is redundant, or `~skore.EstimatorReport(s)` which is confusing to read in raw text.

paulinebanye replied:
That's a good suggestion @thomass-dev 😀. I was torn between using entries, items or objects but instances is great. I'll adjust for both files.

thomass-dev reviewed (changes requested):
Nice catch, thanks!

paulinebanye commented:
Hi @thomass-dev please review. Got E501 errors when running pre-commit, so I removed the "." at the end of instances (in metrics_accessor.py) and moved the "or" to the next line (report.py)

```python
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

skore\src\skore\sklearn\_comparison\metrics_accessor.py:1009:89: E501 Line too long (89 > 88)
     |
1007 |             Function to aggregate the scores across the cross-validation splits.
1008 |             None will return the scores for each split.
1009 |             Ignored when comparison is between :class:`~skore.EstimatorReport` instances.
     |                                                                                         ^ E501
1010 |
1011 |         Returns
     |
```

```python
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

skore\src\skore\sklearn\_comparison\report.py:27:89: E501 Line too long (92 > 88)
   |
25 |     """Report for comparing reports.
26 |
27 |     This object can be used to compare several :class:`~skore.EstimatorReport` instances, or
   |                                                                                         ^^^^ E501
28 |     several :class:`~skore.CrossValidationReport(s)`.
   |
```

thomass-dev reviewed (approved):
Thanks for the fixes!

paulinebanye commented:
> Thanks for the fixes!

@thomass-dev You're welcome 🙂
